lua/cl_spectator_deathmatch.lua

```diff
@@ -264,27 +263,37 @@ hook.Add("Initialize", "Initialize_Ghost", function()
 	end
 
 	function ScoreGroup(p)
-		if not IsValid(p) then return -1 end 
+		if not IsValid(p) then return -1 end
+
+		local group = hook.Call("TTTScoreGroup", nil, p)
+
+		if group then
+			return group
+		end
+
 		local client = LocalPlayer()
-		if p:IsGhost()  then
-			if not p:GetNWBool("PlayedSRound") then return GROUP_SPEC end
+
+		if p:IsGhost() then
+			if not p:GetNWBool("PlayedSRound", false) then
+				return GROUP_SPEC
+			end
+
 			if p:GetNWBool("body_found", false) then
 				return GROUP_FOUND
 			else
-				local client = LocalPlayer()
-				if client:IsSpec() or client:IsActiveTraitor() or ((GAMEMODE.round_state != ROUND_ACTIVE) and client:IsTerror()) then
+				if client:IsSpec() or client:IsActiveTraitor() or ((GAMEMODE.round_state ~= ROUND_ACTIVE) and client:IsTerror()) then
 					return GROUP_NOTFOUND
 				else
 					return GROUP_TERROR
 				end
-			end	   
+			end
 		end
+
 		if DetectiveMode() then
-			if p:IsSpec() and p:GetNWBool("PlayedSRound") and not p:Alive() then
+			if p:IsSpec() and p:GetNWBool("PlayedSRound", false) and not p:Alive() then
```

Comment:
You are alive as GhostDM User, as living player and in some cases as spectator.
It would be possible but why adding this to a hook if the original function shouldn't get called?